src/utils/trading/__tests__/conversion.utils.test.ts

```diff
@@ -1,14 +1,76 @@
+import { ConversionError } from '../../../errors/exchange.errors';
+import { OrderSizeError } from '../../../errors/trading.errors';
+import { getRelativeOrderSize, getTokensAmount } from '../conversion.utils';
+
 describe('Conversion utils', () => {
   describe('getRelativeOrderSize', () => {
-    it.todo('should return relative size');
+    it('should return relative size', () => {
+      expect(getRelativeOrderSize(70, '10%')).toEqual(7);
+      expect(getRelativeOrderSize(70, '100%')).toEqual(70);
+
+      // sad paths
+      expect(getRelativeOrderSize(70, '%50')).toEqual(35);
+      expect(getRelativeOrderSize(70, '%%4%0%%%')).toEqual(28);
+      expect(getRelativeOrderSize(70, 'fifteen percent')).toEqual(NaN);
+      expect(getRelativeOrderSize(70, 'NaN')).toEqual(NaN);
+    });
+
+    it('should throw if range is incorrect', () => {
+      expect(() => getRelativeOrderSize(70, '0%')).toThrowError(OrderSizeError);
+      expect(() => getRelativeOrderSize(70, '100.0000000001')).toThrowError(
+        OrderSizeError
+      );
+      expect(() => getRelativeOrderSize(70, '-0.001%')).toThrowError(
+        OrderSizeError
+      );
+      expect(() => getRelativeOrderSize(70, '%')).toThrowError(OrderSizeError);
 
-    it.todo('should throw if range is incorrect');
+      // sad path
+      expect(() => getRelativeOrderSize(70, null)).toThrowError(TypeError);
+    });
   });
 
   describe('getTokensAmount', () => {
-    it.todo('should return tokens amount');
+    it('should return tokens amount', () => {
+      expect(getTokensAmount('symbol', 50, 20)).toEqual(0.4);
+      expect(getTokensAmount('symbol', 50, 0)).toEqual(0);
 
-    it.todo('should throw on error');
+      // sad paths
+      expect(getTokensAmount('symbol', 50, -150)).toEqual(-3);
+      expect(getTokensAmount('symbol', -50, 150)).toEqual(-3);
+      expect(getTokensAmount('symbol', -50, -150)).toEqual(3);
+      expect(getTokensAmount('symbol', 0, 0.01)).toEqual(Infinity);
+      expect(getTokensAmount('symbol', null, -150)).toEqual(-Infinity);
+      expect(getTokensAmount(null, 3, 45)).toEqual(15);
+    });
+
+    it('should throw on error', () => {
+      expect(() => getTokensAmount(null, 2.8, null)).toThrowError(TypeError);
```

Comment:
Let me know what you think about what direction to go in with this type of test

Reply:
I'm not sure where to go with this, my first intention was to improve existing error handling in order to ensure that we have all we need to do our calculation, you don't really need to test all possible cases, null or NaN should be enough I guess

Reply:
Thanks for the feedback. I'll go lighter moving forward. Feel free to discard anything you like from it